### DISCUSSION TITLE
Clarification regarding a point mentioned in FastAPI's Async Documentation

### DISCUSSION THREAD

**Discussion by hamza-ahmad-emumba** [fastapi/fastapi#10771] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
None
```

### Description

Inside the [FastAPI's Async Documentation](https://fastapi.tiangolo.com/async/) there is a point that states:

`If your application (somehow) doesn't have to communicate with anything else and wait for it to respond, use async def.`

Now after reading through this entire page I think that the recommended choice for the above mentioned point should be `def` instead of `async def`.  Because if there are no I/O Bound operations then `async def` and `def` should work the same and therefore the recommended choice should be `def`.

I am aware that this might be an insignificant thing to open a discussion for but I would greatly appreciate if someone could clarify this.

Thank you for your time.

P.S. Apologies for any mistakes regarding the formatting or structure of this post. 

### Operating System

Windows

### FastAPI Version

None

### Pydantic Version

None

### Python Version

None

**Comment by jgould22** [comment, marked as answer]
FastAPI is an async framework, and async should the be the default choice.

Really the threadpool def endpoints are an escape hatch for running sync/blocking code because the python async ecosystem is still pretty nascent (and sometimes you just need to run blocking code).

There is also the problem of function coloring. You cannot as easily run an async function ( which should be the majority of your functions) inside a def function. 

You also take a performance hit spawning the request to a thread pool.

Anyway I agree that the docs there are worded a bit awkwardly and should probably say something like "If you don't have any blocking IO or CPU bound code you should use async"  but they are supposed to be high level and it gets into it more down the page.
